bundle/src/main/java/com/adobe/acs/commons/httpcache/store/mem/impl/MemTempSinkImpl.java

```diff
@@ -53,7 +53,7 @@ public InputStream createInputStream() throws HttpCacheDataStreamException {
         if (null != sink) {
             return new ByteArrayInputStream(sink);
         } else {
-            throw new HttpCacheDataStreamException("Nothing available in sink.");
```

Comment:
if existing client code makes the assumption that a cachable reponse with `HTTP/1.1 200` but an empty body results in an exception that bubbles up and effectively turns the reponse into a `HTTP/1.1 500`, that assumption would break.

It seems like a particularly unlikely assumption to me, though.

The exception currently occurs for an empty body, as only the call to `getOutputStream()` or `getWriter()` of the cache's respose wrapper creates the underlying `ByteArrayOutputStream` of the in-memory store.

The `TempSink` API contract makes no statement on when to expect the declared `HttpCacheDataStreamException`.